python/src/aac/context/definition_parser.py

```diff
@@ -9,735 +9,841 @@
+                            self.get_location_str(parent_name, definition.lexemes),
+                        )
+
+                    if parent_definition.get_root_key() == "schema":
                         inheritance_parents.append(
-                            context.context_instance.fully_qualified_name_to_class[
-                                parent_fully_qualified_name
-                            ]
+                            self.create_schema_class(parent_definition)
                         )
                     else:
-                        # there is a chance that processing order just means we haven't gotten to the parent yet
-                        parent_definition = None
-                        if (
-                            parent_fully_qualified_name
-                            in fully_qualified_name_to_definition
-                        ):
-                            parent_definition = fully_qualified_name_to_definition[
-                                parent_fully_qualified_name
-                            ]
-
-                        if not parent_definition:
-                            raise LanguageError(
-                                f"Cannot find parent definition {parent_fully_qualified_name} for {definition.name}",
-                                get_location_str(parent_name, definition.lexemes),
-                            )
-
-                        if parent_definition.get_root_key() == "schema":
-                            inheritance_parents.append(
-                                create_schema_class(parent_definition)
-                            )
-                        else:
-                            # AaC only supports schema inheritance
-                            raise LanguageError(
-                                f"AaC extension is only supported for schema.  Unable to create parent class with AaC root: {parent_definition.get_root_key()}",
-                                get_location_str(parent_name, definition.lexemes),
-                            )
-
-            return inheritance_parents
-
-        def create_enum_class(enum_definition: Definition) -> Type:
-            """
-            Creates an enum class from a given enum definition.
-
-            Args:
-                enum_definition (Definition): An enum definition to convert to a class.
-
-            Returns:
-                The created class.
-            """
-            if not enum_definition.get_root_key() == "enum":
+                        # AaC only supports schema inheritance
+                        raise LanguageError(
+                            f"AaC extension is only supported for schema.  Unable to create parent class with AaC root: {parent_definition.get_root_key()}",
+                            self.get_location_str(parent_name, definition.lexemes),
+                        )
+
+        return inheritance_parents
+
+    def create_enum_class(self, enum_definition: Definition) -> Type:
```

Comment:
```suggestion
    def create_enum_class(self, enum_definition: Definition) -> Enum:
```

Because of the comment below about the set to `None`, the IDE gripes that it is in conflict. This should be returning an `Enum` at the end if things are done correctly.